test/distributed/test_multi_threaded_pg.py

```diff
@@ -141,9 +144,34 @@ def test_all_reduce(self):
         res_num = ((0 + self.world_size - 1) * self.world_size) / 2
         self.assertEqual(output, torch.ones(3, 3) * res_num)
 
-        # Test unimplemented error
-        with self.assertRaisesRegex(NotImplementedError, "only supports SUM on threaded pg for now"):
-            dist.all_reduce(output, op=ReduceOp.MAX)
+    def test_all_reduce_ops(self):
+        tensor = torch.tensor([dist.get_rank() + 1])
```

Comment:
nit: op tests below seem like good candidates for [subtests](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests)